ampproject/amphtml
# Support <amp-social-share type="system">

Adds support for the [`navigator.share` origin trial](https://developers.google.com/web/updates/2016/10/navigator-share).

This fixes #5799.

**Screenshots**

`<amp-social-share type="system">`:

![adb-screenshot-20170130153712](https://cloud.githubusercontent.com/assets/51244/22429306/16472182-e702-11e6-9e4b-d489839fdfa8.png)

`<amp-social-share type="system" data-mode="replace">`:

![adb-screenshot-20170130153630](https://cloud.githubusercontent.com/assets/51244/22429330/22fb86ac-e702-11e6-81e7-b56677576a6b.png)

**TODO**

* Design/UI questions:
  * [x] Is `system` the right type? (Another possibility: `native`.)
  * [x] Is the MD "share" icon correct? (Note that at the moment this feature is not supported by Safari, and will not appear on iOS.)
  * [x] What to use as the background color?
* [x] "optionally hide all the other buttons if share is supported"
* [x] [Apply for origin trail token](https://docs.google.com/forms/d/e/1FAIpQLSfO0_ptFl8r8G0UFhT0xhV17eabG-erUWBDiKSRDTqEZ_9ULQ/viewform)

**Testing**

To test you need to be running Chrome on a real mobile device. The next step depends on the domain your AMP is hosted on:

* **localhost**
  1. Enable "Experimental Web Platform features" via <chrome://flags/#enable-experimental-web-platform-features>.
* **Other domains**
  1. [Apply for an origin trial token for your domain](https://goo.gl/GR5YLI).
  1. Insert the tokens on your site. (Note that until #7122 is deployed, `<meta http-equiv="origin-trial">` is invalid AMP, and will not be served by the cache.)

Origin trial token for `cdn.ampproject.org`:

```
AvyXp7EY65ehpmAu/9JOWZVDuLAUiZWk9Tpp9h+lkQuZzednBdNKkbZkOobQkKXKuGonBGLIOP4xqgNsxQY0xwEAAABZeyJvcmlnaW4iOiAiaHR0cHM6Ly9jZG4uYW1wcHJvamVjdC5vcmc6NDQzIiwgImZlYXR1cmUiOiAiV2ViU2hhcmUiLCAiZXhwaXJ5IjogMTQ4ODQ2ODUxMn0=
```
